src/Auth/CredentialProvider.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Momento\Auth;
+
+/**
+ * Provides information that the CacheClient needs in order to establish a connection to and authenticate with
+ * the Momento service.
+ */
+abstract class CredentialProvider
```

Comment:
```suggestion
abstract class CredentialProvider implements ICredentialProvider
```

We'll of course need to add the `ICredentialProvider` file and class back in.